tests/test_data_collections.py

```diff
@@ -26,21 +18,39 @@ def test_derive() -> None:
     assert derived_definition.collection_type is None
 
 
-def test_compare() -> None:
+@pytest.mark.parametrize(
+    "definition_input, expected",
+    [
+        ({}, "DataCollectionDefinition(\n  is_timeless: False\n  has_cloud_coverage: False\n)"),
+        (
+            {"api_id": "X", "_name": "A"},
+            "DataCollectionDefinition(\n  api_id: X\n  is_timeless: False\n  has_cloud_coverage: False\n)",
+        ),
+        (
+            {"api_id": "Y", "is_timeless": True, "has_cloud_coverage": True},
+            "DataCollectionDefinition(\n  api_id: Y\n  is_timeless: True\n  has_cloud_coverage: True\n)",
+        ),
+    ],
+)
+def test_collection_repr(definition_input: Dict[str, Any], expected: str) -> None:
+    assert repr(DataCollectionDefinition(**definition_input)) == expected
+
+
+def test_collection_equality() -> None:
     def1 = DataCollectionDefinition(api_id="X", _name="A")
     def2 = DataCollectionDefinition(api_id="X", _name="B")
 
     assert def1 == def2
 
 
 def test_define() -> None:
-    for _ in range(3):
-        data_collection = DataCollection.define(
-            "NEW", api_id="X", sensor_type="Sensor", bands=("B01",), is_timeless=True
-        )
+    data_collection = DataCollection.define("NEW", api_id="X", sensor_type="Sensor", bands=("B01",), is_timeless=True)
 
     assert data_collection == DataCollection.NEW
+    assert DataCollection.NEW.api_id == "X"
 
+
+def test_define_faild() -> None:
```

Comment:
```suggestion
def test_define_fails() -> None:
```

Reply:
add docstring/comment why this must fail. If you can, perhaps even make the name state the reason for failure, such as `test_define_same_api_id_fails`

Does this fail only because `test_define` gets run before it?

Reply:
add comments for reasons why it should fail

Reply:
Done.